hamt_test.go

```diff
@@ -453,6 +457,12 @@ func TestValueLinking(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	var decodedNode Node
+	cbor.DecodeInto(nd.RawData(), &decodedNode)
+
+	var decodedVal map[string]interface{}
+	cbor.DecodeInto(decodedNode.Pointers[0].KVs[0].Value, &decodedVal)
```

Comment:
I'm still not sure i fully grok the spec, but this appears that all of this is stored inside one cbor node? if thats the case, maybe we need to store the children / grandchildren as their own nodes?

Reply:
256 pointers per node, 3 values per pointer, once the values overflow, a child gets created (and repeat)